src/metax_api/services/catalog_record_service.py

```diff
@@ -760,3 +762,24 @@ def get_research_dataset_license_url(rd):
         license = rd['access_rights']['license'][0]
 
         return license.get('identifier') or license.get('license')
+
+    @classmethod
+    def destroy_bulk(cls, request):
+        """
+        Mark datasets as deleted en masse. Parameter cr_identifiers can be a list of pk's
+        (integers), or file identifiers (strings).
+        """
+        _logger.info('Begin bulk delete datasets')
+
+        cr_ids = cls.identifiers_to_ids(request.data)
+
+        for id in cr_ids:
+            try:
+                cr = CatalogRecord.objects.get(pk=id)
+                if cr.user_has_access(request):
+                    cr.delete()
+            except:
+                pass
+
+        _logger.info(f'Marked datasets {cr_ids} as deleted')
+        return Response(status=status.HTTP_204_NO_CONTENT)
```

Comment:
Should it be feasible to return 200 with list of deleted ids/identifiers if at least one dataset was deleted and 400 if none of the datasets are found or user has no rights to delete them?

Reply:
I was also thinking about returning 200 but noticed that in bulk file delete 200 returned with count of deleted files, which felt not very relevant. But i think you right, 200 is better.